instrumentation/aws_sdk/lib/opentelemetry/instrumentation/aws_sdk/handler.rb

```diff
@@ -0,0 +1,64 @@
+# frozen_string_literal: true
+
+# Copyright The OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+module OpenTelemetry
+  module Instrumentation
+    module AwsSdk
+      # Generates Spans for all interactions with AwsSdk
+      class Handler < Seahorse::Client::Handler
+        def call(context)
+          span_name, attributes = calculate_span(context)
+
+          tracer.in_span(span_name, kind: OpenTelemetry::Trace::SpanKind::CLIENT, attributes: attributes) do |span|
+            execute = proc {
+              super(context).tap do |response|
+                if (err = response.error)
+                  span.record_exception(err)
+                  span.status = Trace::Status.error(err)
+                end
+              end
+            }
+
+            if instrumentation_config[:suppress_internal_instrumentation]
+              OpenTelemetry::Common::Utilities.untraced(&execute)
+            else
+              execute.call
+            end
+          end
+        end
+
+        def calculate_span(context)
+          service_name = context.client.class.api.metadata['serviceId'] || context.client.class.to_s.split('::')[1]
+          span_name = "#{service_name}.#{context.operation.name}"
+          attributes = {
+            'aws.region' => context.config.region,
+            OpenTelemetry::SemanticConventions::Trace::RPC_SYSTEM => 'aws-api',
+            OpenTelemetry::SemanticConventions::Trace::RPC_METHOD => context.operation.name,
+            OpenTelemetry::SemanticConventions::Trace::RPC_SERVICE => service_name
+          }
+
+          [span_name, attributes]
+        end
+
+        def tracer
+          AwsSdk::Instrumentation.instance.tracer
+        end
+
+        def instrumentation_config
+          AwsSdk::Instrumentation.instance.config
+        end
+      end
+
+      # A Seahorse::Client::Plugin that enables instrumentation for all AWS services
+      class Plugin < Seahorse::Client::Plugin
+        def add_handlers(handlers, config)
+          # run before Seahorse::Client::Plugin::ParamValidator (priority 50)
+          handlers.add Handler, step: :validate, priority: 49
```

Comment:
Question for AWS users... is there anyway to look up the position/priority of an existing Handler like we can do with Rails middleware?